workflows-build: update matrix limitation docs (#3523)
AOSC-Dev/aosc-os-abbs: .github/workflows/README.md

- Currently, this command only supports GitHub-hosted runners.
  - Microsoft Azure's Standard_DS2_v2 virtual machines only support x86-64 (`amd64`) architecture and may not have sufficient performance or storage to build some packages.
  - GitHub-hosted runners currently provide no TTY device[^1], any build that relies on this will fail.

> - At most 256 packages can be built at a time.

Reviewer: Up to 256 packages could be built at a time.

Author: ```suggestion
- Up to 256 packages could be built at a time.
```
